core/ecschema-editing/src/Editing/Editor.ts

```diff
@@ -286,3 +291,12 @@ export class SchemaContextEditor {
   }
 }
 
+function isItemType<T extends typeof SchemaItem>(schemaItem: SchemaItem, itemConstructor: T): schemaItem is InstanceType<T> {
```

Comment:
@StefanApfel-Bentley This is sort-of redundant to logic in Schema.ts. I did not think we'd need this outside of there. Maybe we should expose this logic from the metadata package and use it here. Otherwise this will be pretty hard to spot if sometime in the future we expand our SchemaItem types.

Reply:
The logic should probably live on the schema item as this is actually more a question of schemaItemType than schema.